manual/development.md

```diff
@@ -287,6 +287,77 @@ def autocorrect(node)
 end
 ```
 
+### Documentation
+
+Every new cop requires explanation and examples to make it easy for the community
+to understand its purpose. This documentation is generated by `yard` and is added
+directly into the `cop.rb` file. For every `SupportedStyle` and unique
+configuration you have included in the cop, there needs to be examples. Examples must
+have valid Ruby syntax. Do not use upticks.
+
+```ruby
+module Department
+  # Description of your cop. Include description of ALL config options. Particularly
+  # ones that take booleans and arrays, because we generally do not show examples for
+  # configs with these value types.
+  #
+  # @example EnforcedStyle: bar
+  #    # Description about this particular option
+  #
+  #    # bad
+  #    bad_example1
+  #    bad_example2
+  #
+  #    # good
+  #    good_example1
+  #    good_example2
+  #
+  # @example EnforcedStyle: foo (default)
+  #    # Description about this particular option
+  #
+  #    # bad
+  #    bad_example1
+  #    bad_example2
+  #
+  #    # good
+  #    good_example1
+  #    good_example2
+  #
+  # @example AnyUniqueConfigKeyThatIsAString: qux (default)
+  #    # Description about this particular option
+  #
+  #    # bad
+  #    bad_example1
+  #    bad_example2
+  #
+  #    # good
+  #    good_example1
+  #    good_example2
+  #
+  # @example AnyUniqueConfigKeyThatIsAString: thud
+  #    # Description about this particular option
+  #
+  #    # bad
+  #    bad_example1
+  #    bad_example2
+  #
+  #    # good
+  #    good_example1
+  #    good_example2
+  #
```

Comment:
It seems that we need to decide on this blank line.
If we leave a blank line, we should also change the following template.
https://github.com/bbatsov/rubocop/blob/v0.52.1/lib/rubocop/cop/generator.rb#L17-L30

IMO, since the appearance of the document that the user sees does not change, I think that it is good to leave a blank line. In other words, we will change the template.

Reply:
Oops, it has already been changed (It seems I was looking at the old state) 💦